core/src/main/scala/chisel3/internal/Builder.scala

```diff
@@ -344,20 +344,6 @@ private[chisel3] trait HasId extends chisel3.InstanceId {
     case Some(p)          => p.name
     case None             => throwException(s"$instanceName doesn't have a parent")
   }
-  def circuitName: String = _parent match {
```

Comment:
It would probably still be possible to keep this, possibly deprecated. I'm not sure of the value but worth considering--maybe if it comes up as an issue in `7.0.0-RC1` we can restore it.